core/services/relay/evm/evm.go

```diff
@@ -197,9 +197,9 @@ func NewRelayer(ctx context.Context, lggr logger.Logger, chain legacyevm.Chain,
 	sugared := logger.Sugared(lggr).Named("Relayer")
 	mercuryORM := mercury.NewORM(opts.DS)
 	cdcFactory := sync.OnceValues(func() (llo.ChannelDefinitionCacheFactory, error) {
-		chainSelector, err := chainselectors.SelectorFromChainId(chain.ID().Uint64())
-		if err != nil {
-			return nil, fmt.Errorf("failed to get chain selector for chain id %s: %w", chain.ID(), err)
+		chainSelector, err2 := chainselectors.SelectorFromChainId(chain.ID().Uint64())
```

Comment:
sorry, why do you need err2? (and err dose not suffice)

Reply:
Sorry, I understand that it is shadowed but I do not see any problem with this in this code. Does the linter prohibit all shadowing or is there something specifically bad here?

Reply:
The linter often does complain about name shadowing, even for err :(.